subcommand/cert-manager/command.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	if (c.flagCertFile != "" && c.flagKeyFile == "") || (c.flagCertFile == "" && c.flagKeyFile != "") {
+		c.UI.Error(fmt.Sprintf("invalid arguments: %s", errors.New("both tls-cert-file and tls-key-file must be provided")))
+		return 1
+	}
+
+	if c.clientset == nil {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("error loading in-cluster K8S config: %s", err))
+			return 1
+		}
+		c.clientset, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("error creating K8S client: %s", err))
+			return 1
+		}
+	}
+
+	var certSource cert.Source = &cert.GenSource{
+		Name:  "Consul Webhook Certificates",
+		Hosts: strings.Split(c.flagAutoHosts, ","),
+	}
+	if c.flagCertFile != "" {
+		certSource = &cert.DiskSource{
+			CertPath: c.flagCertFile,
+			KeyPath:  c.flagKeyFile,
+		}
+	}
+
+	// Create the certificate notifier so we can update for certificates,
+	// then start all the background routines for updating certificates.
+	certCh := make(chan cert.Bundle)
+	certNotify := &cert.Notify{Ch: certCh, Source: certSource}
+	defer certNotify.Stop()
+	go certNotify.Start(context.Background())
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	c.certWatcher(ctx, certCh, c.clientset, c.UI)
+	return 0
+}
+
+func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.Bundle, clientset kubernetes.Interface, log cli.Ui) {
+	var bundle cert.Bundle
+	for {
+		select {
+		case bundle = <-ch:
+			log.Info("updated certificate bundle received. updating webhook certs.")
+			// Bundle is updated, set it up
+
+		case <-time.After(30 * time.Minute):
+			// This forces the mutating ctrlWebhook config to remain updated
+			// fairly quickly. This is a jank way to do this and we should
+			// look to improve it in the future. Since we use Patch requests
+			// it is pretty cheap to do, though.
+
+		case <-ctx.Done():
+			// Quit
+			return
+		}
+
+		log.Info("getting secret from kubernetes")
+		certSecret, err := clientset.CoreV1().Secrets(c.flagSecretNamespace).Get(ctx, c.flagSecretName, metav1.GetOptions{})
+		if err != nil && k8serrors.IsNotFound(err) {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      c.flagSecretName,
+					Namespace: c.flagSecretNamespace,
+				},
+				Data: map[string][]byte{
+					corev1.TLSCertKey:       bundle.Cert,
+					corev1.TLSPrivateKeyKey: bundle.Key,
+				},
+				Type: corev1.SecretTypeTLS,
+			}
+
+			log.Info("creating kubernetes secret")
+			_, err := clientset.CoreV1().Secrets(c.flagSecretNamespace).Create(ctx, secret, metav1.CreateOptions{})
+			if err != nil {
+				log.Error(fmt.Sprintf("error writing secret to API server: %s", err))
+				continue
+			}
+
+			log.Info("updating webhook configuration with new CA")
+			err = c.updateWebhookConfig(bundle, clientset, ctx)
+			if err != nil {
+				log.Error(fmt.Sprintf("error updating webhook configuration: %s", err))
+				continue
+			}
```

Comment:
Don't we need to `continue` out of here? If we fall through we just update the secret and webhook config again.